savageRoller.ino

```diff
@@ -668,46 +672,59 @@ void splashHandleKeys() {
       displayDeck();
     } else if (isNewlyPressed('b')) {
       currentPage = Page::Battery;
-      firstRun = 1;
+      last_battery_check = 0;
+      batteryHandleDisplay();
     }
   }
 }
 
 /* Display the currently estimated battery percentage on the screen. */
 void batteryHandleDisplay() {
-  if (!stateChange) {
+  // Only update the display if it's been more then 5000 clocks since last check
+  std::clock_t now = std::clock();
+  if (now - last_battery_check < 5000) {
     return;
   }
+  last_battery_check = now;
 
   // Retrieve the current battery level and compute a display string for it.
-  auto level = M5Cardputer.Power.getBatteryLevel();
+  int32_t level = M5Cardputer.Power.getBatteryLevel();
   auto level_label = "Battery: " + String(level) + "%";
 
-  // Start with a blank canvas, of course.
-  M5Cardputer.Display.clear();
-
   // Draw a partially or fully filled in battery in the middle of the screen.
   auto left = displayWidth / 4;
   auto top = displayHeight / 4;
   auto width = displayWidth / 2;
   auto height = displayHeight / 2;
   auto fillwidth = width * level / 100;
   const int16_t radius = 3;
-  M5Cardputer.Display.setColor(LIGHTGREY);
+
+  // Start with a blank canvas, of course.
+  M5Cardputer.Display.clear();
+
+  // Show the text and percentage at the bottom of the screen.
+  M5Cardputer.Display.setTextColor(LIGHTGREY);
+  M5Cardputer.Display.setTextDatum(textdatum_t::bottom_center);
+  M5Cardputer.Display.drawString(level_label, displayWidth/2, displayHeight);
 
   // Draw the general outline of the battery.
   M5Cardputer.Display.drawRoundRect(left, top, width, height, radius);
```

Comment:
You lost the call to `M5Cardputer.Display.setColor(LIGHTGREY);` Without that, you will get whatever color was last used to draw anything. Reinsert that line to make this color selection deterministic.

Reply:
Good catch, I saw `setColor(LIGHTGREY)` and `setTextColor(LIGHTGREY)` and worked too quickly, assuming sameness. Will fix.